src/Memory/InputQueue.test.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import * as util from 'util'
+import * as BB from 'botbuilder'
+import * as CLM from '@conversationlearner/models'
+import { CLState } from './CLState'
+import { InputQueue } from '../Memory/InputQueue'
+
+// InputQueue requires tiny delay between inputs or InputQueue mutex won't fishing setting
+const mutexDelay = 50
+// Delay to simulate CLRunner ProcessInput
+const processDelay = 1000
+const userAccount: BB.ChannelAccount = { id: "user", name: "user", role: BB.RoleTypes.User, aadObjectId: '' }
+const botAccount: BB.ChannelAccount = { id: "bot", name: "bot", role: BB.RoleTypes.Bot, aadObjectId: '' }
+let responses: { [key: string]: string[] } = {}
+
+// Use any as return type as is not a full Activity, just bare minimum for test
+function makeActivity(message: string, conversationId: string): any {
+    return {
+        id: CLM.ModelUtils.generateGUID(),
+        conversation: { id: conversationId },
+        from: botAccount,
+        recipient: userAccount,
+        type: 'message',
+        text: message,
+    }
+}
+
+function sleep(ms: number) {
```

Comment:
would rename to `delay` not sleep, it doesn't block the processer like other `sleep`s